src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParamOrderCheck.java

```diff
@@ -0,0 +1,396 @@
+     *
+     * @param ast the method node.
+     * @return the list of parameter nodes for ast.
+     */
+    private static List<DetailAST> getParameters(DetailAST ast) {
+        final DetailAST params = ast.findFirstToken(TokenTypes.PARAMETERS);
+        final List<DetailAST> returnValue = new ArrayList<>();
+
+        if (params != null) {
+            DetailAST child = params.getFirstChild();
+            while (child != null) {
+            if (child.getType() == TokenTypes.PARAMETER_DEF) {
+                final DetailAST ident = child.findFirstToken(TokenTypes.IDENT);
+                if (ident != null) {
+                    returnValue.add(ident);
+                }
+            }
+            child = child.getNextSibling();
+            }
+        }
+        return returnValue;
+    }
+
+    /**
+     * Returns the tags in a javadoc comment. Only finds throws, exception,
+     * param, return and see tags.
+     *
+     * @param comment the Javadoc comment
+     * @return the tags found
+     */
+    private static List<JavadocTag> getJavadocTags(TextBlock comment) {
+        final String[] lines = comment.getText();
+        final List<JavadocTag> tags = new ArrayList<>();
+        int currentLine = comment.getStartLineNo() - 1;
+        final int startColumnNumber = comment.getStartColNo();
+
+        for (int i = 0; i < lines.length; i++) {
+            currentLine++;
+            final Matcher javadocArgMatcher =
+                MATCH_JAVADOC_ARG.matcher(lines[i]);
+            final Matcher javadocArgMissingDescriptionMatcher =
+                MATCH_JAVADOC_ARG_MISSING_DESCRIPTION.matcher(lines[i]);
+
+            if (javadocArgMatcher.find()) {
+                final int col = calculateTagColumn(javadocArgMatcher, i, startColumnNumber);
+                tags.add(new JavadocTag(currentLine, col, javadocArgMatcher.group(1),
+                        javadocArgMatcher.group(2)));
+            }
+            else if (javadocArgMissingDescriptionMatcher.find()) {
+                final int col = calculateTagColumn(javadocArgMissingDescriptionMatcher, i,
+                    startColumnNumber);
+                tags.add(new JavadocTag(currentLine, col,
+                    javadocArgMissingDescriptionMatcher.group(1),
+                    javadocArgMissingDescriptionMatcher.group(2)));
+            }
+        }
+        return tags;
+    }
+
+    /**
+     * Calculates column number using Javadoc tag matcher.
+     *
+     * @param javadocTagMatcher found javadoc tag matcher
+     * @param lineNumber line number of Javadoc tag in comment
+     * @param startColumnNumber column number of Javadoc comment beginning
+     * @return column number
+     */
+    private static int calculateTagColumn(Matcher javadocTagMatcher,
+            int lineNumber, int startColumnNumber) {
+        int col = javadocTagMatcher.start(1) - 1;
+        if (lineNumber == 0) {
+            col += startColumnNumber;
+        }
```

Comment:
```
[WARNING] Rule violated for class com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocParamOrderCheck: lines covered ratio is 0.98, but expected minimum is 1.00
[WARNING] Rule violated for class com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocParamOrderCheck: branches covered ratio is 0.88, but expected minimum is 1.00
```
because of these lines